xarray/core/dataarray.py

```diff
@@ -1385,6 +1385,13 @@ def reindex(
         align
         """
         indexers = either_dict_or_kwargs(indexers, indexers_kwargs, "reindex")
+        if isinstance(fill_value, dict):
+            fill_value = fill_value.copy()
+            sentinel = object()
+            value = fill_value.pop(self.name, sentinel)
+            if value is not sentinel:
+                fill_value["<this-array>"] = value
```

Comment:
we usually use `_THIS_ARRAY = ReprObject("<this-array>")` for this kind of thing. But I don't know if that's really appropriate / needed here.

Reply:
I think we need that because the name of the temp dataset is `_THIS_ARRAY`, not the name of the DataArray:
```suggestion
                fill_value[_THIS_ARRAY] = value
```